k8s/pkg/apis/longhorn/v1beta2/objectstore.go

```diff
@@ -0,0 +1,232 @@
+package v1beta2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ObjectStoreState string
+
+const (
+	ObjectStoreStateUnknown     = ObjectStoreState("unknown")
+	ObjectStoreStateStarting    = ObjectStoreState("starting")
+	ObjectStoreStateRunning     = ObjectStoreState("running")
+	ObjectStoreStateStopping    = ObjectStoreState("stopping")
+	ObjectStoreStateStopped     = ObjectStoreState("stopped")
+	ObjectStoreStateTerminating = ObjectStoreState("terminating")
+	ObjectStoreStateError       = ObjectStoreState("error")
+)
+
+type ObjectStoreStatus struct {
```

Comment:
How should that field be updated when the manager goes away?

As far as I can tell, the `controllerID` field of other managers is populated with the node's hostname. But if a node crashes and goes away for a longer time, this would prevent another manager from picking up the `ObjectStore` and reconciling it. This could cause an `ObjectStore` to be unable to transition into `stopped` or `running` or `terminating` state while a node is unavailable as it won't be reconciled again until the node comes back. In addition to that, in a K8s cluster it is not guaranteed that a node ever comes back, e.g. when the cluster is dynamically scaled. This could cause the manager of this object store to disappear permanently.

Reply:
Sounds good to me that you did
>  to tie the ObjectStore and the volume together, so whichever Longhorn manager is responsible for the volume is also going to be responsible for the ObjectStore

Marking this one as resolved